src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
```

Comment:
Assuming the above comments, we should refactor this line to something like this:

    return totalWoman/women.length;  
We don't need to check if totalWoman is a number, because the reduce method will return the initial value (in our case is 0) if the reducer function returns undefined.

It's not considered a good practice, but i recommend storing this calculation in a variable, to achieve a more readable code.